### ISSUE TITLE
Clicking by a mouse or trackpad cannot focus the cursor

### ISSUE THREAD

**Issue by kkk669** [ish-app/ish#731]
Tapping the screen can focus the cursor, but clicking by a mouse or trackpad can't do.

![IMB_Kau7FZ](https://user-images.githubusercontent.com/601636/82726826-9c4d1380-9d21-11ea-8a3c-9ceb568fb505.gif)

I'm using 11-inch iPad Pro (2020) running iPadOS 13.5 and iSH 1.0 (67).

**Comment by repl-mike-roest** [comment]
Also noticed that highlighting text works with the cursor but you can't then command-c to copy (not sure if its related but it's also cursor related)

**Comment by kkk669** [comment]
Fixed in build 72.

**Comment by repl-mike-roest** [comment]
I'm still able to reproduce this in build 72

**Comment by kkk669** [comment]
@repl-mike-roest Which problem are you saying about? The problem I reported at first has been fixed in my environment.

**Comment by repl-mike-roest** [comment]
Both issues the copy/paste I noted but I’m still having your original posted issue as well.

**Comment by kkk669** [comment]
@repl-mike-roest It's weird. Possibly, it's because I'm using iPadOS 14 Developer Beta 1. I reopened this issue. Thanks.

**Comment by repl-mike-roest** [comment]
Thanks!

**Comment by FrederickGeek8** [comment]
I’m having this problem on build 73. Worse still, if I click with my mouse and defocus the cursor, then I’m unable to refocus it with the touchscreen, Apple Pencil, and mouse.

**Comment by puthi** [comment]
I’m facing the same issue on iPad Pro 11 2018 (iPadOS 14.1) and iSH 1.0.1, using external mouse and external keyboard.
As side note, sometimes double click on the small rectangle, helps to bring back the focus. And when it lost focus you can actually Ctrl+C or Ctrl +D, it will receive the key press but you cannot type any text.

**Comment by notDavid** [comment]
Hi, so I have the same issue.

FYI the workaround when this happens is to click (not tap) the trackpad/cursor and keep the "mouse button" down while clicking for a second.
After that, you can type again (and the bar in the window below will also reappear.)

IPadOS 14.6
ISH 1.2 (build 149) - from the AppStore

**Comment by kgsbowtie** [comment]
So the issue is still there in 1.2.2 (Build 178), but the [mentioned workaround](https://github.com/ish-app/ish/issues/731#issuecomment-850527521) above works (holding down the trackpad). Another workaround I’ve been using is single tap on the trackpad and that seems to always work as well.

What’s interesting though is that I while testing that workaround I found that if you don’t hold down the trackpad/mouse button long enough, the keyboard/typing focus is always lost exactly when the trackpad is release. 

So maybe it has to do with some JavaScript `mouseup` event? Or the speed between the `mousedown` and `mouseup` events maybe? I don’t know exactly, but some rudimentary testing of `MouseEvent` performance, it seems tapping (either on the screen or trackpad) is about 1ms between `mousedown` and `mouseup` and clicking on the trackpad (the fastest I could do it) was about 30ms.

Dunno, none of this may matter and it may not be related to this. Just trying to add information to help track things down to fix it.
—
Magic Keyboard
iPadOS 14.7.1
iSH 1.2.2 (Build 178) - from AppStore

**Comment by thorolason** [comment]
I have the same issue here using an external bluetooth keyboard with a touchpad. Using version 1.3.2 which is the lastest one her in feb 2024.  Unfortunatelly none of the workarounds are working for me so only possibility is to kill the app and start again.  no alphanumeric char works .. so fx the arrows work you can scroll through latest commands and ctrl-c works for newline but no new text can by typed and backspace also not working.   Started therefore to use another program a-shell for ssh sessions. but would really like this to be fixed.. Any ide .. is it possible to just disable the trackpad when in the shell .. i dont really need it actually, but the fingers just slipp there occationally and then booom.. have to kill the app.

**Comment by adamrdrew** [comment]
I’m also experiencing an inability to type after the app loses and then regains focus. I’m on iSH 1.3.2 build 494 on the latest iPad OS 17. In this state the cursor turns into a box. Shortcut keys work, arrow keys work, cut and paste works, and scroll works but letter keys, space, delete etc do not work. The issue is somewhat sporadic. I can use the app for a while switching back and forth fine but then at some point I’ll return focus to iSH and input wont work anymore. Also, the on screen menu disappears in this state too. I tried every possible key combo I could think of, including command comma to open space, which does work, but doesn’t result in text input working again. None of the work arounds in this thread worked.

**Comment by thorolason** [comment]
The best “workaround” i have found is to disable the “tap to click” feature
in the trackpad.  but that is then for all apps.  but keeps this
from happening accidentally in ish.

**Comment by jaedub8** [comment]
Having the exact same issue as @adamrdrew 
IPad Pro M4, IpadOS 17.5, iSH 1.3.2 (build 494)

For me it mostly seems to occur when I use the mouse to select text I want to copy.  But it also occurs when the app is running in the background and I switch to it.  Oddly, it doesn’t happen every time for both scenario, but maybe 90% of the time I select text with the mouse.  Holding down the trackpad does not function as a workaround me, so I’m forced to close the app, which is frustrating for my workflows because I lose my place and have to start over from the beginning.  It’s a real productivity killer for my own workflows so hope this can get higher priority.  Love iSH though!  Thanks!

**Comment by drpump** [comment]
+1. Disabling “tap to click” as suggested by @thorolason seems to work so far, but I’d prefer to have it enabled. I have the latest iOS and latest app store version of iSH.

Note that I have a similar issue with `code-server` when running in a Safari window, so I suspect there is some strange behaviour in iPadOS. The problem has occurred both with my new ESR keyboard/trackpad, and also when using a separate bluetooth keyboard and USB mouse.

**Comment by the-hampel** [comment]
+1 have the same problem. No way to regain focus in iSH. Running the latest version from the AppStore with keyboard and touchpad. Any help welcome.

**Comment by kkkonrad** [comment]
same problem, disabling tap to click not working, external bluetooth keyboard and touchpad connected

**Comment by nohr** [comment]
i found that using the Apple Pencil restores input as a quick workaround but would still like to see this issue fixed

**Comment by kkkonradaws** [comment]
> i found that using the Apple Pencil restores input as a quick workaround but would still like to see this issue fixed

yes, restore input works also on Apple Pencil cheap alternatives

**Comment by kkkonrad** [comment]
> i found that using the Apple Pencil restores input as a quick workaround but would still like to see this issue fixed

yes, restore input works also on Apple Pencil cheap alternatives

**Comment by notinachos** [comment]
+1 I am also having this problem.

iPad Pro M4 13"
Apple Magic Keyboard
iPadOS 18.1
latest version of ish from app store

**Comment by schainks** [comment]
The cursor focus works in the bare sense of the word for Stage Manager users on iPad OS if you go to Settings -> Appearance -> Always Hide Status Bar -> Disable.

Your UI will be janky b/c the bar disappears when the app is out of focus, and comes back when in focus, but it does ensure cursor focus returns properly. I am using my iPad with an external monitor and stage manager.

**Comment by jaedub8** [comment]
Whether “Always Hide Status Bar” is Disabled or Enabled, it seems to break just the same for me.  I mentioned before as well, but to reconfirm it’s still the case, it breaks most often for me while trying to select and copy text.. pretty easy to reproduce too.

**Comment by TmacP** [comment]
This issue is affecting me too.
IPadOS 18.4.1 with Magic Trackpad. iSH version 1.3.2

**Comment by ww2283** [comment]
I have the same issue; following up for updates.

**Comment by wang20150419** [comment]
I also encountered this issue and look forward to a solution.
